tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,89 @@
+```Python
+import cornac
+from cornac.data import Reader
+from cornac.datasets import movielens
+
+plots, movie_ids = movielens.load_plot()
+rating_data = movielens.load_100k(reader=Reader(item_set=movie_ids, bin_threshold=3))
+```
+where we have filtered out movies without plots and binarized the integer ratings thanks to the `Reader`.
+
+### The Text Modality class
+
+With our dataset in place, the next step is to instantiate a TextModality, which will allow us to manipulate and represent our auxiliary data in the desired format.  
+ ```Python
+from cornac.data import TextModality
+from cornac.data.text import BaseTokenizer
+
+item_text_modality = TextModality(corpus=plots, ids=movie_ids, 
+                                  tokenizer=BaseTokenizer(sep='\t', stop_words='english'),
+                                  max_vocab=5000, max_doc_freq=0.5)
+```
+In addition the movies plots and ids, we have specified a tokenizer to split text, we limited the maximum vocabulary size to 5000, as well as filtered out words occurring in more than 50% of documents (plots in our case) by setting `max_doc_freq = 0.5`. For more options/details please refer to the [docs](https://cornac.readthedocs.io/en/latest/data.html#module-cornac.data.text). 
+ 
+ 
+**Bag-of-Word text representation.** CDL assumes the bag-of-word representation for text information, i.e., in the form of a document-word matrix. The good news is that we don't have to worry about how to generate such a representation from our raw texts. The TextModality class implements the necessary routines to process and output different representations for text data, e.g., sequence, bag of word. That is, to get our auxiliary data under the desired format, all we need inside the CLD code `cornac/models/cdl/recom_cld` is the following line of code:
+```Python
+text_feature = self.train_set.item_text.batch_bow(np.arange(n_items))
+``` 
+where `self.train_set.item_text` is set to our instantiated text modality `item_text_modality`, `n_items` is number of training items, and the `batch_bow()` function returns the bag-of-word vectors of the specified item ids, in our case we want the text features for all training items: `np.arange(n_items)`.
+
+**Aligning auxiliary information with the set of training items.** Another important aspect worth mentioning at this level is that, we don't have to take extra actions to align the set of training movies with their plots. They are already aligned. That is, the first row in `text_feature` corresponds to training `movie_id: 1`, the second row to training `movie_id: 2`, and so on. This is made possible thanks to passing `item_text_modality` through the evaluation (splitting) method as we shall see shortly. 
```

Comment:
This is expected behavior, and I've already added a sentence to explain it earlier.  I think this will confuse the readers. Can we remove it?

Reply:
Thanks for this comments. I still want to highlight this aspect. I will change it into a note, make it much shorter (1 or 2 sentence), and refer to the previous text.

Reply:
In that case, there are two things that we need to make clear and consistent. First, when passing into an evaluation method, users and items will go through the indexing process where each of them will be assigned a unique `index`, not `id` (to avoid confusion when referring to their original ids). Second, the user/item indices start counting from 0, not 1.

Reply:
That's fine. Can you please check the new version and let me if its ok. There is no reference to id anymore. I don't want to bring in too much details either, as this can make things harder to follow.